activerecord/lib/active_record/model_schema.rb

```diff
@@ -595,6 +603,18 @@ def load_schema!
           super
         end
 
+        def _run_load_schema_callbacks # :nodoc:
+          @_load_schema_callback_already_called ||= false
+          yield
+          __after_load_schema_callbacks.each do |(callback_context, block)|
+            callback_context.instance_exec(&block)
```

Comment:
MVP implementation. If we get the shape right, the details don't matter. (Except that it's necessarily different from all other callbacks e.g. in activerecord/lib/active_record/callbacks.rb )

I wonder if it makes sense to use an ActiveSupport.on_load hook

Reply:
> I wonder if it makes sense to use an ActiveSupport.on_load hook

That's a good question. I suppose in most cases you don't really care about a particular model being loaded. That said, in development, models are loaded one by one, so a global hook wouldn't work.

So I think we have no choice but to do this per model :s

Reply:
Thanks @casperisfine for taking a look. I should have responded sooner that I'm behind on returning to this to review what's next.